api/pkg/apis/projectcalico/v3/felixconfig.go

```diff
@@ -344,20 +344,20 @@ type FelixConfigurationSpec struct {
-	// For back-compatibility, if the protocol is not specified, it defaults to "tcp". If a CIDR is not specified, it will allow
-	// traffic from all addresses. To disable all inbound host ports, use the value none. The default value allows ssh access
-	// and DHCP.
-	// [Default: tcp:22, udp:68, tcp:179, tcp:2379, tcp:2380, tcp:6443, tcp:6666, tcp:6667]
+	// FailsafeInboundHostPorts is a list of PortProto struct objects including UDP/TCP/SCTP ports and CIDRs that Felix will
+	// allow incoming traffic to host endpoints on irrespective of the security policy. This is useful to avoid accidentally
+	// cutting off a host with incorrect configuration. For backwards compatibility, if the protocol is not specified,
+	// it defaults to "tcp". If a CIDR is not specified, it will allow traffic from all addresses. To disable all inbound host ports,
+	// use the value "[]". The default value allows ssh access, DHCP, BGP, etcd and the Kubernetes API.
+	// [Default: {"port":22,"protocol":"tcp"},{"port":68,"protocol":"udp"},{"port":179,"protocol":"tcp"},{"port":2379,"protocol":"tcp"}, {"port":2380,"protocol":"tcp"}, {"port":5473,"protocol":"tcp"}, {"port":6443,"protocol":"tcp"}, {"port":6666,"protocol":"tcp"}]
 	FailsafeInboundHostPorts *[]ProtoPort `json:"failsafeInboundHostPorts,omitempty"`
-	// FailsafeOutboundHostPorts is a list of UDP/TCP ports and CIDRs that Felix will allow outgoing traffic from host endpoints
-	// to irrespective of the security policy. This is useful to avoid accidentally cutting off a host with incorrect configuration.
-	// For back-compatibility, if the protocol is not specified, it defaults to "tcp". If a CIDR is not specified, it will allow
-	// traffic from all addresses. To disable all outbound host ports, use the value none. The default value opens etcd's standard
-	// ports to ensure that Felix does not get cut off from etcd as well as allowing DHCP and DNS.
-	// [Default: tcp:179, tcp:2379, tcp:2380, tcp:6443, tcp:6666, tcp:6667, udp:53, udp:67]
+	// FailsafeOutboundHostPorts is a list of List of PortProto struct objects including UDP/TCP/SCTP ports and CIDRs that Felix
+	// will allow outgoing traffic from host endpoints to irrespective of the security policy. This is useful to avoid accidentally
+	// cutting off a host with incorrect configuration. For backwards compatibility, if the protocol is not specified, it defaults
+	// to "tcp". If a CIDR is not specified, it will allow traffic from all addresses. To disable all outbound host ports,
+	// use the value "[]". The default value opens etcd's standard ports to ensure that Felix does not get cut off from etcd
+	// as well as allowing DHCP, DNS, BGP and the Kubernetes API.
+	// [Default: {"port":53,"protocol":"udp"},{"port":67,"protocol":"udp"}, {"port":179,"protocol":"tcp"}, {"port":2379,"protocol":"tcp"}, {"port":2380,"protocol":"tcp"}, {"port":5473,"protocol":"tcp"}, {"port":6443,"protocol":"tcp"}, {"port": 6666,"protocol":"tcp"}, {"port":6667,"protocol":"tcp"}]
```

Comment:
Is this really how it should be specified? Or should it be udp:53 and then itis parsed into the struct? HOw would you specify it in envvar?

Reply:
Judging by the test cases, I would say my approach is correct. 
https://github.com/projectcalico/calico/blob/fa5e216cd7af37c594a224d0afb070d3ccea0ff9/felix/config/config_params_test.go#L385C1-L389C6

Reply:
That's the Golang representation, but the yaml representation might be different. 

Not sure the right one to put here. Probably should be representation agnostic?

Reply:
Note this part of the test cases you linked:

> "tcp:0.0.0.0/0:1,udp:0.0.0.0/0:2"

^ This is the input, which is then parsed into the structs.